src/stash/src/postgres.rs

```diff
@@ -628,12 +643,13 @@ impl Stash {
                 // can't accidentally have the same epoch, nonce pair (especially risky if the
                 // current epoch has been bumped exactly once, then gets recreated by another
                 // connection that also bumps it once).
-                let row = tx
-                    .query_one(
-                        "UPDATE fence SET epoch=epoch+1, nonce=$1 RETURNING epoch",
-                        &[&self.nonce.to_vec()],
-                    )
-                    .await?;
+                let update = match epoch_lower_bound {
+                    Some(epoch_lower_bound) => {
+                        format!("UPDATE fence SET epoch=GREATEST(epoch+1, {}), nonce=$1 RETURNING epoch", epoch_lower_bound.get())
```

Comment:
very optional: could map None to the minimum Epoch to only have one query here

Reply:
Done